geomstats/geometry/hyperbolic_space.py

```diff
@@ -374,13 +389,28 @@ def dist(self, point_a, point_b):
         dist : array-like, shape=[n_samples, 1]
                            or shape=[1, 1]
         """
-        sq_norm_a = self.embedding_metric.squared_norm(point_a)
-        sq_norm_b = self.embedding_metric.squared_norm(point_b)
-        inner_prod = self.embedding_metric.inner_product(point_a, point_b)
 
-        cosh_angle = - inner_prod / gs.sqrt(sq_norm_a * sq_norm_b)
-        cosh_angle = gs.clip(cosh_angle, 1.0, 1e24)
+        if(self.point_type==None):
+
+            sq_norm_a = self.embedding_metric.squared_norm(point_a)
+            sq_norm_b = self.embedding_metric.squared_norm(point_b)
+            inner_prod = self.embedding_metric.inner_product(point_a, point_b)
+
+            cosh_angle = - inner_prod / gs.sqrt(sq_norm_a * sq_norm_b)
+            cosh_angle = gs.clip(cosh_angle, 1.0, 1e24)
+
+            dist = gs.arccosh(cosh_angle)
+
+            return dist
+
+        if(self.point_type=='Poincare'):
+
+            point_a_norm = gs.clamp(gs.sum(point_a ** 2, dim=-1), 0, 1 - 1e-3)
```

Comment:
`gs.sum(point_a ** 2, dim=-1` do you mean the square norm?
`gs.linalg.norm(point_a) ** 2` would maybe be more readable?

Reply:
what is the difference between `clamp` and `clip` functions?

Reply:
Introduce a constant: `EPSILON = 1e-3` at the beginning of the file to make the numerical aspect more clear.

Reply:
> `gs.sum(point_a ** 2, dim=-1` do you mean the square norm?
> `gs.linalg.norm(point_a) ** 2` would maybe be more readable?

We think the first line is faster in terms of execution, however the second one is more readable

Reply:
> Introduce a constant: `EPSILON = 1e-3` at the beginning of the file to make the numerical aspect more clear.

Excellent idea